File: src/test/java/org/example/page/SearchAllPage.java
package org.example.page;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

import java.util.List;

public class SearchAllPage {

    public WebDriver driver;

    public SearchAllPage(WebDriver driver) {
        PageFactory.initElements(driver, this);
        this.driver = driver;
    }

    @FindBy(xpath = "//*[@class='pagesubheader' and text()='Все продукты']")
    private WebElement searchAllPageText;

    @FindBy(xpath = "//div[@data-loc='SteamOS + Linux']")
    private WebElement steamOSAndLinux;

    @FindBy(xpath = "//div[@data-loc='Кооператив (LAN)']")
    private WebElement coopLan;

    @FindBy(xpath = "//div[@data-loc='Экшен']")
    private WebElement action;

    @FindBy(xpath = "//div[@data-collapse-name='os']")
    private WebElement os;

    @FindBy(xpath = "//div[@data-collapse-name='category3']")
    private WebElement countPlayers;

    @FindBy(xpath = "//div[@data-collapse-name='tags']")
    private WebElement tags;

    @FindBy(xpath = "//div[@id='search_resultsRows']/child::a")
    private List<WebElement> foundedGames;

    public String getSearchAllPageText() {
        return searchAllPageText.getText();
    }

    public void enableSteamOSAndLinux() {
        String value = os.getAttribute("class");
        List<String> list = List.of(value.split(" "));
        boolean checkClassName = list.contains("collapsed");
        if (checkClassName) {
            os.click();
        }
        if (!steamOSAndLinux.isSelected()) {
            steamOSAndLinux.click();
        }
    }

    public boolean existSteamOSAndLinux() {
        return steamOSAndLinux.isSelected();
    }

    public void enableCoopLan() {
        String value = countPlayers.getAttribute("class");
        List<String> list = List.of(value.split(" "));
        boolean checkClassName = list.contains("collapsed");
        if (checkClassName) {
            countPlayers.click();
        }
        if (!coopLan.isSelected()) {
            coopLan.click();
        }
    }

    public boolean existCoopLan() {
        return coopLan.isSelected();
    }

    public void enableAction() {
        String value = tags.getAttribute("class");
        List<String> list = List.of(value.split(" "));
        boolean checkClassName = list.contains("collapsed");
        if (checkClassName) {
            tags.click();
        }
        if (!action.isSelected()) {
            action.click();
        }
    }

    public boolean existAction() {
        return action.isSelected();
    }

    public String getGamesInfo() {
        enableSteamOSAndLinux();
        enableCoopLan();
        enableAction();
        return foundedGames.get(0).getText();
    }
}

File: src/test/java/org/example/page/SearchAllPage.kt
package org.example.page

import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement
import org.openqa.selenium.support.FindBy
import org.openqa.selenium.support.PageFactory

class SearchAllPage(driver: WebDriver) {
    private var driver: WebDriver

    @FindBy(xpath = "//*[@class='pagesubheader' and text()='Все продукты']")
    private val searchAllPageText: WebElement? = null

    @FindBy(xpath = "//div[@data-loc='SteamOS + Linux']")
    private val steamOSAndLinux: WebElement? = null

    @FindBy(xpath = "//div[@data-loc='Кооператив (LAN)']")
    private val coopLan: WebElement? = null

    @FindBy(xpath = "//div[@data-loc='Экшен']")
    private val action: WebElement? = null

    @FindBy(xpath = "//div[@data-collapse-name='os']")
    private val os: WebElement? = null

    @FindBy(xpath = "//div[@data-collapse-name='category3']")
    private val countPlayers: WebElement? = null

    @FindBy(xpath = "//div[@data-collapse-name='tags']")
    private val tags: WebElement? = null

    @FindBy(xpath = "//div[@id='search_resultsRows']/child::a")
    private val foundedGames: List<WebElement>? = null

    init {
        PageFactory.initElements(driver, this)
        this.driver = driver
    }

    fun getSearchAllPageText(): String {
        return searchAllPageText!!.text
    }

    fun enableSteamOSAndLinux() {
        val value = os!!.getAttribute("class")
        val list = java.util.List.of(*value.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray())
        val checkClassName = list.contains("collapsed")
        if (checkClassName) {
            os.click()
        }
        if (!steamOSAndLinux!!.isSelected) {
            steamOSAndLinux.click()
        }
    }

    fun existSteamOSAndLinux(): Boolean {
        return steamOSAndLinux!!.isEnabled
    }

    fun enableCoopLan() {
        val value = countPlayers!!.getAttribute("class")
        val list = java.util.List.of(*value.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray())
        val checkClassName = list.contains("collapsed")
        if (checkClassName) {
            countPlayers.click()
        }
        if (!coopLan!!.isSelected) {
            coopLan.click()
        }
    }

    fun existCoopLan(): Boolean {
        return coopLan!!.isEnabled
    }

    fun enableAction() {
        val value = tags!!.getAttribute("class")
        val list = java.util.List.of(*value.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray())
        val checkClassName = list.contains("collapsed")
        if (checkClassName) {
            tags.click()
        }
        if (!action!!.isSelected) {
            action.click()
        }
    }

    fun existAction(): Boolean {
        return action!!.isEnabled
    }

    val gamesInfo: String
        get() {
            enableSteamOSAndLinux()
            enableCoopLan()
            enableAction()
            return foundedGames!![0].text
        }
}